src/main/java/com/aws/greengrass/util/GreengrassServiceClientFactory.java

```diff
@@ -54,13 +59,15 @@ public GreengrassServiceClientFactory(DeviceConfiguration deviceConfiguration) {
             }
             if (validString(node, DEVICE_PARAM_AWS_REGION) || validString(node, DEVICE_PARAM_ROOT_CA_PATH)
                     || validString(node, DEVICE_PARAM_CERTIFICATE_FILE_PATH) || validString(node,
-                    DEVICE_PARAM_PRIVATE_KEY_PATH) || validString(node, DEVICE_PARAM_GG_DATA_PLANE_PORT)) {
-                validateConfiguration();
+                    DEVICE_PARAM_PRIVATE_KEY_PATH) || validString(node, DEVICE_PARAM_GG_DATA_PLANE_PORT)
+                    || validString(node, DEVICE_PARAM_IOT_CRED_ENDPOINT) || validString(node,
+                    DEVICE_PARAM_IOT_DATA_ENDPOINT)) {
+                logger.atDebug().kv("modifiedNode", node.getFullName()).kv("changeType", what)
+                        .log("Queued re-validation of Greengrass v2 data client.");
+                deviceConfigChanged.set(true);
```

Comment:
couple nits:
1. could this spam a bunch of logs? Would it be better to do something like:

```
logger.atTrace().kv("what", what).kv("node", node.getFullName()).log()
if (deviceConfigChanged.compareAndSet(false, true)) {
  logger.atDebug().log("Queued re-validation of Greengrass v2 data client.");
}
```

2. Don't add the "." at the end, otherwise we get "client.." in the logs

Reply:
will update